Nil pointer crash in memberlist wrapper

Starting store-gateway with a problem which causes it to exit immediately:

```
level=info ts=2020-12-21T10:32:06.957805157Z caller=module_service.go:90 msg="module stopped" module=runtime-config
panic: interface conversion: interface {} is nil, not *memberlist.KV

goroutine 36 [running]:
github.com/cortexproject/cortex/pkg/ring/kv/memberlist.(*KVInitService).getKV(...)
        /go/src/github.com/cortexproject/cortex/pkg/ring/kv/memberlist/kv_init_service.go:67
github.com/cortexproject/cortex/pkg/ring/kv/memberlist.(*KVInitService).stopping(0xc00056ee00, 0x0, 0x0, 0x2b4f340, 0xc0000aebe0)
        /go/src/github.com/cortexproject/cortex/pkg/ring/kv/memberlist/kv_init_service.go:81 +0xea
github.com/cortexproject/cortex/pkg/util/services.(*BasicService).main(0xc0004f1a70)
        /go/src/github.com/cortexproject/cortex/pkg/util/services/basic_service.go:184 +0x2fc
created by github.com/cortexproject/cortex/pkg/util/services.(*BasicService).StartAsync.func1
        /go/src/github.com/cortexproject/cortex/pkg/util/services/basic_service.go:96 +0xa8
```

I'll create a PR to make `getKV()` do what its comment says.

--- a/pkg/ring/kv/memberlist/kv_init_service.go
+++ b/pkg/ring/kv/memberlist/kv_init_service.go
@@ -64,7 +64,11 @@ func (kvs *KVInitService) GetMemberlistKV() (*KV, error) {
 
 // Returns KV if it was initialized, or nil.
 func (kvs *KVInitService) getKV() *KV {
-	return kvs.kv.Load().(*KV)
+	kv := kvs.kv.Load()
+	if kv == nil {
+		return nil
+	}
+	return kv.(*KV)
 }
 
 func (kvs *KVInitService) running(ctx context.Context) error {

--- a/pkg/ring/kv/memberlist/kv_init_service_test.go
+++ b/pkg/ring/kv/memberlist/kv_init_service_test.go
@@ -50,3 +50,9 @@ func TestPage(t *testing.T) {
 		}},
 	}))
 }
+
+func TestStop(t *testing.T) {
+	var cfg KVConfig
+	kvinit := NewKVInitService(&cfg, nil)
+	require.NoError(t, kvinit.stopping(nil))
+}

